src/App.tsx

```diff
@@ -16,80 +19,171 @@ export const goodsFromServer = [
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
-
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
-    </div>
-  );
-};
+export class App extends Component<{}, State> {
+  state = {
+    sortType: SortType.NONE,
+    isReversed: false,
+  };
+
+  goodsReverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sortByAlphabet = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  resetButton = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  render() {
+    const { sortType, isReversed } = this.state;
+
+    const preparedGoods = (
+      getReorderedGoods(goodsFromServer, { sortType, isReversed })
+    );
+
+    return (
+      <>
+        <div className="container">
+          <Box
+            sx={{
+              display: 'flex',
+              '& > :not(style)': {
+                m: 1,
+                mt: 30,
+                mx: 'auto',
+                width: 665,
+                height: 500,
+              },
+            }}
+          >
+            <Paper
+              variant="outlined"
+              sx={{
+                backgroundColor: 'text.disabled',
+                display: 'flex',
+                alignItems: 'center',
+                justifyContent: 'start',
+                fontSize: 24,
+                fontWeight: 'bold',
+              }}
+            >
+              <div className="section content">
+                <div className="buttons">
+                  <Button
+                    variant="contained"
+                    type="button"
+                    className={sortType === SortType.ALPHABET
+                      ? 'button is-info'
+                      : 'button is-info is-light'}
+                    onClick={this.sortByAlphabet}
```

Comment:
you can also do it with imported classNames, I think it can save your time) but your solution also good as for me)